docs/workshops/advanced/node-workflow.md

```diff
@@ -9,54 +9,54 @@ begin
 
 ```python
-def get_nodes_list() -> List[Any]:
-    """
-    Connects to netbox and returns a list of netbox device objects.
-    """
-    logger.debug("Connecting to Netbox to get list of available nodes")
-    node_list = list(netbox.dcim.devices.filter(status="planned"))
-    logger.debug("Found nodes in Netbox", amount=len(node_list))
-    return node_list
+def initial_input_form_generator(product_name: str) -> FormGenerator:
+    """Generates the Node Form to display to the user."""
+    logger.debug("Generating initial input form")
+    nodes = netbox.get_devices(status="planned")
+    choices = [node.name for node in nodes]
+    NodeEnum = Choice("Planned nodes", zip(choices, choices))  # type: ignore
 ```
 
-This function is quite straightforward and simply connects to netbox and gets a list of all devices that are in the `planned` state using the netbox client. Since we don't have anything in the workflow that then changes that status in netbox from planned to a different state, there is nothing stopping us from enrolling the same router over and over again, which is of course not desirable.
+The important part of this function here is `nodes = netbox.get_devices(status="planned")`, where we fetch all devices in netbox that are in the `planned` state using the netbox service and then we construct the rest of the initial input form to present a list of available nodes to run this workflow against. Since we don't have anything in the workflow that then ever changes that status in netbox from planned to a different state, there is nothing stopping us from enrolling the same router over and over again, which is of course not desirable.
 
 ### Implement The Update Step
 
-The `pynetbox` client we are using has a very convenient way of updating items in netbox. If you are familiar with Django (the underlying technology for netbox) or other ORMs the client will feel very familiar to you. The general process is to fetch an object, modify it locally, then save your changes. Thus, to make the update happen in netbox, we need to do a three step process:
-
-1. Use the netbox client to retrieve the device.
-2. Update the device status.
-3. Save the device changes in netbox.
+The example netbox service provided with this workshop code is very convenient for updating data in netbox via the orchestrator. The general way this works, is that we have pre-written dataclasses that match the expected netbox API payloads, and made an orchestrator service that you can pass the subscription object to. Once the netbox service receives the subscription object, all you need to do is update the domain model with the right value, and then the code in the `update_node_in_netbox` step of the node create workflow will call the netbox service and update things appropriately in that system. To take advantage of this service, all you need to do is the following in the `set_node_to_active` step of the node workflow:
```

Comment:
We might want to explain the general idea behind the single dispatch call: The first argument to `build_payload()` is the product block instance to indicate the specific payload we want to build, the second argument is the complete subscription because sometimes other information, like general subscription attributes or information from other product blocks that are part of this subscription, to build the payload.

Reply:
Awesome, updated!